app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/ResourceMetaDiffUtilCallback.kt

```diff
@@ -0,0 +1,23 @@
+package space.taran.arknavigator.mvp.presenter.adapter
+
+import androidx.recyclerview.widget.DiffUtil
+import space.taran.arknavigator.mvp.model.repo.index.ResourceMeta
+
+class ResourceMetaDiffUtilCallback(
+    private val oldList: List<ResourceMeta>,
+    private val newList: List<ResourceMeta>
+) : DiffUtil.Callback() {
+    override fun getOldListSize(): Int = oldList.size
+
+    override fun getNewListSize(): Int = newList.size
+
+    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int):
+        Boolean {
+        return oldList[oldItemPosition] == newList[newItemPosition]
```

Comment:
Why comparing by reference and not by value?

Reply:
In Kotlin `==` compares by calling `equals` under the hood. To compare by reference you need to use `===`

Reply:
Oh, indeed :)